setup.py

```diff
@@ -56,8 +56,8 @@ def read_requirements(filename):
     "smdebug_rulesconfig==1.0.1",
     "importlib-metadata>=1.4.0,<7.0",
     "packaging>=20.0",
-    "pandas",
     "pathos",
+    "pytz",
```

Comment:
do we need this new dependency, how much of a file size addition is this? if needed can we rely on existing dependencies (/some dependency from python standard library)

Reply:
https://github.com/aws/sagemaker-python-sdk/blob/a4ba730b906f1b15504da471157ac6af513ebde4/src/sagemaker/feature_store/feature_processor/feature_scheduler.py#L21

it's used in the code base but is installed implicitly through `pandas`. I wanted to address it in a separate PR. But can also do it here

Reply:
Yeah we should really really careful while adding new dependency and only do it if absolutely necessary, otherwise use extras [added dependency]

Reply:
Pandas < 2.0 necessitates `pytz` but [Pandas>=2.0 now uses `datetime.timezone` by default](https://pandas.pydata.org/docs/dev/whatsnew/v2.0.0.html#utc-and-fixed-offset-timezones-default-to-standard-library-tzinfo-objects).

Probably `pytz` was introduced when installing Pandas < 2.0.

If we depends on Pandas 2.0, we can safely remove `pytz` as a constraint.